front/lib/resources/user_resource.ts

```diff
@@ -0,0 +1,167 @@
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class UserResource extends BaseResource<User> {
+  static model: ModelStatic<User> = User;
+
+  constructor(model: ModelStatic<User>, blob: Attributes<User>) {
+    super(User, blob);
+  }
+
+  static async makeNew(
+    blob: Omit<
+      Attributes<User>,
+      | "id"
+      | "createdAt"
+      | "updatedAt"
+      | "isDustSuperUser"
+      | "providerId"
+      | "imageUrl"
+    > &
+      Partial<Pick<Attributes<User>, "providerId" | "imageUrl">>
+  ): Promise<UserResource> {
+    const user = await User.create(blob);
+    return new this(User, user.get());
+  }
+
+  static async fetchAllByModelIds(ids: number[]): Promise<UserResource[]> {
+    const users = await User.findAll({
+      where: {
+        id: ids,
+      },
+    });
+
+    return users.map((user) => new UserResource(User, user.get()));
+  }
+
+  static async fetchAllByUsername(username: string): Promise<UserResource[]> {
+    const users = await User.findAll({
+      where: {
+        username,
+      },
+    });
+
+    return users.map((user) => new UserResource(User, user.get()));
+  }
+
+  static fetchAllByEmail(email: string): Promise<UserResource[]> {
+    return User.findAll({
+      where: {
+        email,
+      },
+    }).then((users) => users.map((user) => new UserResource(User, user.get())));
+  }
+
+  static async fetchByExternalId(userId: string): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        sId: userId,
+      },
+    });
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async fetchByAuth0Sub(sub: string): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        auth0Sub: sub,
+      },
+    });
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async fetchByEmail(email: string): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        email,
+      },
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async fetchByProvider(
+    provider: UserProviderType,
+    providerId: string
+  ): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        provider,
+        providerId,
+      },
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async getWorkspaceFirstAdmin(
+    workspaceId: number
+  ): Promise<UserResource | null> {
+    const user = await User.findOne({
+      include: [
+        {
+          model: MembershipModel,
+          where: {
+            role: "admin",
+            workspaceId,
+          },
+          required: true,
+        },
+      ],
+      order: [["createdAt", "ASC"]],
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  async update(blob: Partial<Attributes<User>>): Promise<void> {
```

Comment:
We should not have a generic method like this, IIRC there is only one place were we update the users, it's in the login flow, let have a dedicated function here.